### ISSUE TITLE
GoogleMaps iOS: initialRegion latitudeDelta/longitudeDelta not respected

### ISSUE THREAD

**Issue by lachenmayer** [react-community/react-native-maps#2100]
### Is this a bug report?

Yes. Please see the repo containing a minimal example: https://github.com/lachenmayer/react-native-maps-google-maps-initialregion-bug

### Observations

When using Google Maps with react-native-maps on iOS, the `initialRegion` prop is not respected.

The map is centered on the correct region, but the zoom level is incorrect, ie. the `latitudeDelta` & `longitudeDelta` values are ignored.

I am aware of the issue https://github.com/react-community/react-native-maps/issues/1903 and the related PR https://github.com/react-community/react-native-maps/pull/1953 which claims to fix this.

Strangely, when I edit the `initialRegion` of one of the maps in the example project (which I was able to get running using the changes in https://github.com/react-community/react-native-maps/pull/2094 - would be great to get this merged), it works as expected.

Because I was not able to reproduce this in the example project, I created a minimal test case by deleting all unrelated code from the project in which I'm experiencing this.

### Environment

| product               | version                                                                                              |
| --------------------- | ---------------------------------------------------------------------------------------------------- |
| react-native          | 0.54.2                                                                                               |
| react-native-maps     | https://github.com/react-community/react-native-maps/commit/6cc87aa5bf25f10ca1c689eed0ddb2c2e551a933 |
| Xcode                 | 9.0                                                                                                  |
| iOS                   | 11.0 (Simulator)                                                                                     |
| GoogleMaps (Pod)      | 2.5.0                                                                                                |
| Google-Maps-iOS-Utils | 2.1.0                                                                                                |

### To reproduce

1. Download the repo & open the Xcode workspace (_not_ the xcodeproj).
2. Run the project in the iOS simulator.
3. The resulting app should show a full-screen map.

_Note: This repo contains all of the required dependencies (node_modules & pods). You don't need to run `yarn/npm` or `pod install`._

### Expected behavior

You should see London:

<img width="483" alt="expected" src="https://user-images.githubusercontent.com/38614/37475397-853dee90-286a-11e8-8b5f-def80ea9a920.png">

You can trigger this behavior by changing `initialRegion` in `src/App.js` to `region`:

```diff
<View style={styles.container}>
  <MapView
    provider={PROVIDER_GOOGLE}
    style={styles.map}
-    initialRegion={{
+    region={{
      latitude: 51.50853,
      longitude: -0.12574,
      latitudeDelta: 1,
      longitudeDelta: 1,
    }}
  />
</View>
```

### Actual behavior

The map is zoomed out:

<img width="483" alt="actual" src="https://user-images.githubusercontent.com/38614/37475377-7d28b442-286a-11e8-9201-ab0730ac58df.png">

**Comment by lachenmayer** [comment]
Hi there, I have found a simple one-line fix to this, you can see it on https://github.com/lachenmayer/react-native-maps-google-maps-initialregion-bug/tree/debugging along with a detailed discussion of the root cause.

The actual fix is this:

```diff
 - (void)setInitialRegion:(MKCoordinateRegion)initialRegion {
  NSLog(@"setInitialRegion start: initial[latd:%f lond:%f] region[latd:%f lond:%f]", _initialRegion.span.latitudeDelta, _initialRegion.span.longitudeDelta,  _region.span.latitudeDelta, _region.span.longitudeDelta);
  if (_initialRegionSetOnLoad) return;
  _initialRegion = initialRegion;
-  _initialRegionSetOnLoad = true;
+  _initialRegionSetOnLoad = _didCallOnMapReady;
  self.camera = [AIRGoogleMap makeGMSCameraPositionFromMap:self andMKCoordinateRegion:initialRegion];
  NSLog(@"setInitialRegion end: initial[latd:%f lond:%f] region[latd:%f lond:%f]", _initialRegion.span.latitudeDelta, _initialRegion.span.longitudeDelta,  _region.span.latitudeDelta, _region.span.longitudeDelta);
}
```

This fixes this bug, as well as resolving the flicker mentioned in https://github.com/react-community/react-native-maps/pull/1950. It also preserves the behavior mentioned in https://github.com/react-community/react-native-maps/pull/1953#issuecomment-356268221, ie. that `initialRegion` should only be set once.

Since this seems a little counter-intuitive I'll probably add a comment explaining this in my actual PR.

@danielgindi & @Agontuk, it would be amazing if you could take a look at the repo above & let me know if there are any issues with it - you seem to be most familiar with this bug.

Caveat: I have not tested this with Android, but since there are no JS changes, it should hopefully not change anything.

**Comment by lachenmayer** [comment]
If you are affected by this, try using the https://github.com/lachenmayer/react-native-maps/tree/initialregion branch - will do a PR once I get some feedback.

**Comment by Agontuk** [comment]
I have not yet faced this problem on my end, lat/lng delta is working fine. So I can't tell for certain whether it's a setup issue or an actual issue. If the example project is working correctly, then it's probably a setup problem. It would be better if you're able to reproduce this somehow.

**Comment by magichim** [comment]
me, too. same problem on ios.

**Comment by magichim** [comment]
@lachenmayer thanks to your fixing, I can solve this problem. What's the reason and Could you explain it?

**Comment by lachenmayer** [comment]
@Agontuk - please check the example repo https://github.com/lachenmayer/react-native-maps-google-maps-initialregion-bug
I don't think this is a setup problem.

**Comment by lachenmayer** [comment]
@magichim I have tried to explain the issue here https://github.com/lachenmayer/react-native-maps-google-maps-initialregion-bug/tree/debugging
I'm not sure why it does not trigger in the example repo.

**Comment by magichim** [comment]
@lachenmayer your example repo has a this error.

<img width="333" alt="2018-03-18 3 46 41" src="https://user-images.githubusercontent.com/5843652/37563427-b9af25d0-2ac3-11e8-9650-e64e58ec4a50.png">

I open trippin.xcworkspace on xcode. and then build and run simulator.
but there are errors like image.

**Comment by lachenmayer** [comment]
@magichim - thanks for checking this out! :)
I just tried to delete the repo & check it out from scratch - it works for me.
Are you definitely sure a packager is running? Did you start the packager using `yarn start`?

**Comment by magichim** [comment]
@lachenmayer 
I just git download and then directly build in xcode.

**Comment by lachenmayer** [comment]
I can confirm that this happens on every map when using React Native 0.54, even the maps in the example app. Check out https://github.com/react-community/react-native-maps/pull/2115

**Comment by wbattel4607** [comment]
We're experiencing this as well on RN 0.54.

**Comment by wbattel4607** [comment]
@lachenmayer I tried your fix and it works great. (only tested iOS). I hope this gets published soon!

**Comment by Agontuk** [comment]
Seems like it's only happening in RN 0.54, I also encountered this after upgrading.

**Comment by Agontuk** [comment]
@lachenmayer, your solution works fine. But I think it should be `_initialRegionSetOnLoad = _didMoveToWindow;` instead of `_initialRegionSetOnLoad = _didCallOnMapReady;`. Can you check if it works ?

**Comment by lachenmayer** [comment]
That also works - why do you prefer this?

**Comment by Agontuk** [comment]
Because in iOS, map is ready long before the `onMapReady` event is triggered. So we can set the initial region right away when view moved into the window.

**Comment by lachenmayer** [comment]
Makes sense! I've made a PR with this change https://github.com/react-community/react-native-maps/pull/2131.

**Comment by rborn** [comment]
@lachenmayer merged #2131 , closing. 🤗

**Comment by keung7251** [comment]
I have already install the old one, how can I upgrade it? I am using react native 0.53
